heaviside/activities.py

```diff
@@ -183,68 +186,122 @@ def fanout(session, sub_sfn, sub_args, max_concurrent=50, rampup_delay=15, rampu
         ActivityError : If there was a failure in a sub_sfn execution, contains
                         the failure error and cause if it can be determined.
     """
+    args = {
+        'sub_sfn': sub_sfn,
+        'sub_args': list(sub_args), # Handle generator type that could previous be passed
+        'max_concurrent': max_concurrent,
+        'rampup_delay': rampup_delay,
+        'rampup_backoff': rampup_backoff,
+        'status_delay': status_delay,
+        'finished': False,
+        'running': [],
+        'results': []
+    }
+
+    while True:
+        args = fanout_nonblocking(args, session)
+
+        if args['finished']:
+            return args['results']
+
+        time.sleep(poll_delay)
+
+def fanout_nonblocking(args, session=None):
+    """Helper method for executing a dynamic number of StepFunctions and
+    returning the results.
+
+    This is a non-blocking version of fanout, designed to be executed by
+    a Lambda in a StepFunction loop, where the loop handles the poll_delay
+    sleep. The function is designed so the output is the input for the next
+    iteration of the polling loop.
+
+    Args:
+        args: {
+            sub_sfn (ARN)
+            sub_args (list)
+            max_concurrent (int)
+            rampup_delay (int)
```

Comment:
Can you put time units in for the delays?

Reply:
Added time units to fanout documentation and added a note to look at fanout documentation for details about shared arguments.